base/gfsptqueue/queue_test.go

```diff
@@ -0,0 +1,185 @@
+package gfsptqueue
+
+import (
+	"testing"
+
+	storagetypes "github.com/bnb-chain/greenfield/x/storage/types"
+	"github.com/stretchr/testify/require"
+
+	"github.com/bnb-chain/greenfield-storage-provider/base/types/gfsptask"
+	"github.com/bnb-chain/greenfield-storage-provider/core/task"
+)
+
+func TestApprovalTaskRetireByExpiredHeight(t *testing.T) {
+	task1 := &gfsptask.GfSpCreateObjectApprovalTask{
+		CreateObjectInfo: &storagetypes.MsgCreateObject{
+			ObjectName:        "test_task_1",
```

Comment:
moved these unit test to testCases struct may be better.

Reply:
the task pointer is be used, inner struct can not accquie the pointer